### DISCUSSION TITLE
Kafka producer doesn't have metrics show in Jconsole

### DISCUSSION THREAD

**Discussion by skas-zilo** [strimzi/strimzi-kafka-operator#10162] [Q&A]
Hi
I'm using Strimzi version 0.34.0. I'm using jmxOptions: {} to enabled jmx port 9999. Port is enabled. When I open Jconsole to see kafka.producer domain. I don't see anythings related. I found that my env in kafka-cluster is STRIMZI_JMX_ENABLED instead of KAFKA_JMX_ENABLED. I'm trying to research but not see topics that related and don't know how different. Is this related to my problem? I would like to get kafka.producer. Can someone suggest me on this issue?

![Screenshot 2567-05-20 at 14 12 00](https://github.com/strimzi/strimzi-kafka-operator/assets/135793639/fe5ccb85-d48b-4c97-a96a-774c22479aac)

<img width="420" alt="Screenshot 2567-05-27 at 11 31 02" src="https://github.com/strimzi/strimzi-kafka-operator/assets/135793639/ccd96ce4-32a3-4a09-83d0-9d33b3df8303">

**Comment by katheris** [comment]
Hey, the `kafka.producer` metrics are reported by producer clients, not by Kafka brokers or controllers. The metrics collected and exposed by Strimzi are the ones reported by the Kafka brokers and controllers. You need to attach jconsole to your producer application
